plugins/utils/hardtime.nix

```
@@ -1,244 +1,228 @@
+    '';
+
+    notification = defaultNullOpts.mkBool true ''
+      Enable notification messages for restricted and disabled keys.
+    '';
+
+    allow_different_key = defaultNullOpts.mkBool false ''
+      Allow different keys to reset the count.
+    '';
+
+    enabled = defaultNullOpts.mkBool true ''
+      Whether the plugin in enabled by default or not.
+    '';
+
+    resetting_keys = defaultNullOpts.mkAttrsOf (with types; listOf str) {
+      "1" = [
+        "n"
+        "x"
+      ];
+      "2" = [
+        "n"
+        "x"
+      ];
+      "3" = [
+        "n"
+        "x"
+      ];
+      "4" = [
+        "n"
+        "x"
+      ];
+      "5" = [
+        "n"
+        "x"
+      ];
+      "6" = [
+        "n"
+        "x"
+      ];
+      "7" = [
+        "n"
+        "x"
+      ];
+      "8" = [
+        "n"
+        "x"
+      ];
+      "9" = [
+        "n"
+        "x"
+      ];
+      "c" = [ "n" ];
+      "C" = [ "n" ];
+      "d" = [ "n" ];
+      "x" = [ "n" ];
+      "X" = [ "n" ];
+      "y" = [ "n" ];
+      "Y" = [ "n" ];
+      "p" = [ "n" ];
+      "P" = [ "n" ];
+    } "Keys in what modes that reset the count.";
+
+    restricted_keys = defaultNullOpts.mkAttrsOf (with types; listOf str) {
+      "h" = [
+        "n"
+        "x"
+      ];
+      "j" = [
+        "n"
+        "x"
+      ];
+      "k" = [
+        "n"
+        "x"
+      ];
+      "l" = [
+        "n"
+        "x"
+      ];
+      "-" = [
+        "n"
+        "x"
+      ];
+      "+" = [
+        "n"
+        "x"
+      ];
+      "gj" = [
+        "n"
+        "x"
+      ];
+      "gk" = [
+        "n"
+        "x"
+      ];
+      "<CR>" = [
+        "n"
+        "x"
+      ];
+      "<C-M>" = [
+        "n"
+        "x"
+      ];
+      "<C-N>" = [
+        "n"
+        "x"
+      ];
+      "<C-P>" = [
+        "n"
+        "x"
+      ];
+    } "Keys in what modes triggering the count mechanism.";
+
+    restriction_mode =
+      defaultNullOpts.mkEnumFirstDefault
+        [
+          "block"
+          "hint"
+        ]
+        ''
+          The behavior when `restricted_keys` trigger count mechanism.
+        '';
+
+    disabled_keys = defaultNullOpts.mkAttrsOf (with types; listOf str) {
+      "<Up>" = [
+        ""
+        "i"
+      ];
+      "<Down>" = [
+        ""
+        "i"
+      ];
+      "<Left>" = [
+        ""
+        "i"
+      ];
+      "<Right>" = [
+        ""
+        "i"
+      ];
+    } "Keys in what modes are disabled.";
+
+    disabled_filetypes = defaultNullOpts.mkListOf types.str [
+      "qf"
+      "netrw"
+      "NvimTree"
+      "lazy"
+      "mason"
+    ] "`hardtime.nvim` is disabled under these filetypes.";
+
+    hints =
+      lib.nixvim.mkNullOrOption
+        (
+          with types;
+          attrsOf (submodule {
+            options = {
+              message = lib.mkOption {
+                description = "Hint message to be displayed.";
+                type = lib.nixvim.nixvimTypes.rawLua;
```

```suggestion
                type = types.rawLua;
```

or should this be mkStrLuaFn ?

---

Actually, you have `with types;` above it
```suggestion
                type = rawLua;
```

---

Let's not introduce more `strLua` into the codebase unnecessarily.

If the option were not declared, the freeform definition would require `__raw`, so I think it best that any declared option do this too.